### PR TITLE
Pin max `dask` and `distributed` versions to `2021.09.1`

### PR DESCRIPTION
**Pull request by galipremsagar**
Changes to be in-line with: https://github.com/rapidsai/cudf/pull/9286

### CODE REVIEW

File: conda/recipes/dask-cuda/meta.yaml [hunk 1]
```diff
@@ -27,8 +27,8 @@ requirements:
     - setuptools
   run:
     - python
-    - dask >=2.22.0
-    - distributed >=2.22.0
+    - dask >=2.22.0,<=2021.09.1
+    - distributed >=2.22.0,<=2021.09.1
```
Comment (pentschev) [inline conda/recipes/dask-cuda/meta.yaml:31]: I think we should pin to _exactly_ `=2021.09.1`. There have been changes that are breaking in older versions, e.g. https://github.com/rapidsai/dask-cuda/pull/722 .

**Review by pentschev — CHANGES_REQUESTED** [review]

File: conda/recipes/dask-cuda/meta.yaml (hunk 1 above)
Comment (pentschev) [reply conda/recipes/dask-cuda/meta.yaml:31]: Any objections @jakirkham @quasiben to my comment above?

Comment (jakirkham) [reply conda/recipes/dask-cuda/meta.yaml:31]: Am happy to defer to you here Peter :)

Comment (pentschev) [reply conda/recipes/dask-cuda/meta.yaml:31]: Let's then pin to `=2021.09.1`, I don't see any compelling reason to support half-functional versions.

Could you do that @galipremsagar ? And before I forget, we also need to set https://github.com/rapidsai/dask-cuda/blob/28ce42701fdac4bb1f06ca7705885e8f1f657c0b/ci/gpu/build.sh#L36 to `0`. Same with other repos.

Comment (galipremsagar) [reply conda/recipes/dask-cuda/meta.yaml:31]:
```suggestion
    - dask=2021.09.1
    - distributed=2021.09.1
```

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-dask>=2.22.0
-distributed>=2.22.0
+dask>=2.22.0,<=2021.09.1
+distributed>=2.22.0,<=2021.09.1
```
Comment (galipremsagar) [inline requirements.txt:2]:
```suggestion
dask==2021.09.1
distributed==2021.09.1
```

**Review by pentschev — APPROVED** [review]
LGTM, thanks @galipremsagar !

**Comment by pentschev** [comment]
@gpucibot merge

**Comment by pentschev** [comment]
I believe we need someone from ops to merge, could you please do that @ajschmidt8 ?

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**